pkg/snapstore/abs_snapstore.go

```diff
@@ -122,10 +130,9 @@ func (a *ABSSnapStore) List() (SnapList, error) {
 
 		// Process the blobs returned in this result segment
 		for _, blob := range listBlob.Segment.BlobItems {
-			k := blob.Name[len(a.prefix)+1:]
-			s, err := ParseSnapshot(k)
+			s, err := ParseSnapshot(path.Join(prefix, blob.Name))
```

Comment:
Shouldn't we have the `HasPrefix` check here?

Reply:
Don't know how did I miss it!